addons/website_slides/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,94 @@
+        all_partner_ids = (src_partners + dst_partner).ids
+        all_channels = SlideChannelPartner_sudo.search([('partner_id', 'in', all_partner_ids)]).channel_id
+        user_partner_id_lst = [(user, user.partner_id.id)
+                               for user in User_sudo.search([('partner_id', 'in', all_partner_ids)])]
+
+        # Save karma to support the karma correction after the recompute of the completion
+        init_earned_karma_by_partner_id = all_channels._get_earned_karma(all_partner_ids)
+        user_karma = {user: user.karma for (user, _) in user_partner_id_lst}
+
+        super()._merge_internal(src_partners, dst_partner)
+
+        # Recompute the completion
+        for slide_partner_relation in SlidePartner_sudo.search([('partner_id', '=', dst_partner.id)]):
+            slide_partner_relation._recompute_completion()
+        SlideChannelPartner_sudo.search([('partner_id', '=', dst_partner.id)])._recompute_completion()
+
+        # Karma correction: Karma has been impacted incorrectly by the recompute of the completion
+        earned_karma = all_channels._get_earned_karma([dst_partner.id])[dst_partner.id]
+        for user, partner_id in user_partner_id_lst:
+            delta_karma = earned_karma - init_earned_karma_by_partner_id.get(partner_id, 0)
+            user.karma = user_karma[user] + delta_karma
+
+    @api.model
+    def _merge_partners_for_unique_m2m(self, src_partners, dst_partner,
+                                       res_model, other_record_field, order=None, prefer_dst_partner=False):
+        """ Merge records of res_model for the given partners following the given preferences.
+
+        ! It is the caller responsibility to check the ACL, this method is executed in sudo.
+        Use this method when the model ("join" model that links uniquely partner to another
+        model) contains a unique key per partner (foreign key to the other model), to select
+        the records to assign to the dst_partner from the record of src_partners and dst_partner.
+        Indeed, when there is an uniq constraint on partner_id and another field we cannot simply
+        assign all record of src_partners to dst_partner without potentially create conflicts
+        (unique constraint violation).
+        When multiple records with the same unique key per partner among the record of the
+        given partners (src_partners and dst_partner) are encountered, this method choose
+        the one with the highest preference and assign it to dst_partner. Of course, if there
+        is only one, it is just assigned to the dst_partner.
+
+        :param recordset src_partners: source partners
+        :param recordset dst_partner: destination partner
+        :param str res_model: many-to-many relationship model linking res_partner and another
+            model through the other_record_field
+        :param str other_record_field: field name of the unique "other" record per partner
```

Comment:
the name `other_record_field` can be more explicit I think, maybe `uniq_field` or something else ? (would like to know, just by reading the name, that we will keep only 1 record / value of this field if you see what I mean)